diff --git a/share/cmake/modules/Findpystring.cmake b/share/cmake/modules/Findpystring.cmake
index 7b894a45..92618215 100644
--- a/share/cmake/modules/Findpystring.cmake
+++ b/share/cmake/modules/Findpystring.cmake
@@ -113,6 +113,11 @@ if(NOT pystring_FOUND)
             -DCMAKE_INSTALL_MESSAGE=${CMAKE_INSTALL_MESSAGE}
             -DCMAKE_INSTALL_PREFIX=${_EXT_DIST_ROOT}
             -DCMAKE_OBJECT_PATH_MAX=${CMAKE_OBJECT_PATH_MAX}
+            -DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}
+            -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
+            -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
+            -DCMAKE_CXX_FLAGS_DEBUG=${CMAKE_CXX_FLAGS_DEBUG}
+            -DCMAKE_CXX_FLAGS_RELEASE=${CMAKE_CXX_FLAGS_RELEASE}
         )
         if(CMAKE_TOOLCHAIN_FILE)
             set(pystring_CMAKE_ARGS

diff --git a/src/OpenColorIO/SystemMonitor.cpp b/src/OpenColorIO/SystemMonitor.cpp
index 329f8673..0e178e50 100644
--- a/src/OpenColorIO/SystemMonitor.cpp
+++ b/src/OpenColorIO/SystemMonitor.cpp
@@ -28,1 +28,1 @@
-#elif __APPLE__
+#elif __APPLE__ && !__IOS__

diff --git a/src/OpenColorIO/CMakeLists.txt b/src/OpenColorIO/CMakeLists.txt
index 7f5b1f29..27b98b4c 100644
--- a/src/OpenColorIO/CMakeLists.txt 
+++ b/src/OpenColorIO/CMakeLists.txt 
@@ -234,5 +227,13 @@ if (APPLE)
+	if("${CMAKE_SYSTEM_NAME}" STREQUAL "iOS")
+		target_compile_definitions(OpenColorIO PRIVATE __IOS__)
+		target_link_libraries(OpenColorIO
+			PRIVATE
+				"-framework IOKit"
+		)
+	else()
-	target_link_libraries(OpenColorIO
-		PRIVATE
-			"-framework Carbon"
-			"-framework IOKit"
-	)
+		target_link_libraries(OpenColorIO
+			PRIVATE
+				"-framework Carbon"
+				"-framework IOKit"
+		)
+	endif()
